### ISSUE TITLE
Improve Qute self-closing tags support

### ISSUE THREAD

**Issue by fbricon** [xorye/vscode-quarkus#596]
In a Qute template, if you start typing `{#inclu` and select completion,  the following is inserted:
```
{#include template}

{/include} 
```
It'd be nice to remove the closing tag if the user adds the self-closing slash after the template name (`{#include template /}`), while the tag content is empty, akin to self-closing tag behavior in vscode-xml.

There might be other specific tags where auto-close might be relevant (#insert? #fragment? ...), @mkouba or @FroMage may have some ideas

**Comment by FroMage** [comment]
That's a good idea, and valid for every tag.

**Comment by fbricon** [comment]
@FroMage I doubt this would be wanted in the case of control flow statements (`{#if}`, `{#for}` ...)

**Comment by FroMage** [comment]
Well, those always have bodies. But all the tags that have a `/` to close them without body need to have their closing tag removed.

**Comment by angelozerr** [comment]
> @FroMage I doubt this would be wanted in the case of control flow statements ({#if}, {#for} ...)

Indeed, we ned to check if section tags has nested content  (for user tag the html file includes {nested-content, we use already this information for completion with user tag)

I think it would be nice too to process the remove of `/`

When we remove this character, it should generate the end tag section.
